monai/transforms/io/dictionary.py

```diff
@@ -143,6 +159,24 @@ def __call__(self, data, reader: Optional[ImageReader] = None):
                 if meta_key in d and not self.overwriting:
                     raise KeyError(f"Meta data with key {meta_key} already exists and overwriting=False.")
                 d[meta_key] = data[1]
+
+        # Only warn the first time per sample to prevent spamming the logs
+        if self.warn_on_shape_mismatch and (first_path not in self.has_warned_about):
+            # all() is a lazy function, i.e. it will stop at the first instance of shapes mismatching
+            if not all(x == image_shapes[0] for x in image_shapes):
+                warnings.warn(
+                    message="Loaded image shapes do not match for input dictionary:"
+                    + f"\n{input_dict}"
+                    + "\nResulting in image shapes:"
+                    + "\n".join(f"\t{key}: {value.shape}" for key, value in d.items())
```

Comment:
```suggestion
                    + "\n".join(f"\t{key}: {value.shape}" for key, value in d.items() if hasattr(value, "shape"))
```
this needs additional checks for the attribute, something like:
```suggestion
                    + "\n".join(f"\t{key}: {value.shape}" for key, value in d.items() if hasattr(value, "shape"))
```

Reply:
I've opted for a slightly different solution that will print `"Not an Image"` whenever there is no `shape` attribute. On reflection, maybe `"No shape available"` might've been more appropriate/general.